plugin/src/main/java/at/helpch/chatchat/util/FormatUtils.java

```diff
@@ -40,25 +38,37 @@ private FormatUtils() {
         return format.parts().stream()
                 .map(part -> PlaceholderAPI.setPlaceholders(player, part))
                 .map(part -> part.replace("%message%", message))
-                .map(part -> replaceRecipientPlaceholder(player, part))
                 .map(FormatUtils::parseToMiniMessage)
                 .collect(Component.toComponent());
     }
 
+    public static @NotNull Component parseFormat(
+        @NotNull final Format format,
+        @NotNull final Player player,
+        @NotNull final Player receiver,
+        @NotNull final String message) {
+        return format.parts().stream()
+            .map(part -> PlaceholderAPI.setPlaceholders(player, part))
+            .map(part -> part.replace("%message%", message))
+            .map(part -> replaceRecipientPlaceholder(receiver, part))
+            .map(FormatUtils::parseToMiniMessage)
+            .collect(Component.toComponent());
+    }
+
     public static @NotNull Component parseToMiniMessage(@NotNull final String formatPart) {
         return miniMessage.deserialize(formatPart, TagResolver.standard());
     }
 
     private static @NotNull String replaceRecipientPlaceholder(@NotNull final Player player, @NotNull final String toReplace) {
-        if (toReplace.equalsIgnoreCase("%recipient%")) {
-            return player.getName();
-        }
-
-        if (toReplace.length() <= RECIPIENT_SUBSTRING) {
-            return toReplace; // prevents IndexOutOfBoundsException from String#substring
+        if (!toReplace.contains("%recipient")) {
+            return toReplace;
         }
 
-        //set any PAPI placeholders after %recipient_
-        return PlaceholderAPI.setPlaceholders(player, "%" + toReplace.substring(RECIPIENT_SUBSTRING));
+        return PlaceholderAPI.setPlaceholders(
+            player,
+            toReplace
+                .replace("%recipient%", player.getName())
+                .replace("%recipient_", "%")
```

Comment:
This replaces partial placeholders with just a single percent sign?

Reply:
Yes. that's exactly what it does. I know we'd need a more complex parser for this but I don't have the time, skills or patience to do that. Someone else is welcomed to do if they want. I just wanted to get something that works even tho not perfect.

Reply:
I don't really get the point of doing that. Is that so that it can parse placeholders on the other side for the recipient?

Reply:
that could probably just be .replace("recipient_", "") tbh

Reply:
%recipient_PAPI-PLACEHOLDER%. That's the whole point of the method

Reply:
I was thinking about that Kaliber but I think this is better. the chances of `recipient_` being part of another placeholder or something like that is way much higher than `%recipient_`

Reply:
Maybe put a comment for that?

Reply:
A comment for what exactly?

Reply:
Oh for what it does? Yeah I guess it makes sense. Its not my method. I just updated it. Will do tho.

Reply:
Done.